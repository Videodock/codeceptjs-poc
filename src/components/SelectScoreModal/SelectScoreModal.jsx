import React from "react";
import PropTypes from 'prop-types';
import { Box, Container } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';

import SelectScoreItem from "../SelectScoreItem/SelectScoreItem";
import styles from './SelectScoreModal.styles';

const useStyles = makeStyles(styles);

export default function SelectScoreModal({ composition, onLoadScore }){
  const classes = useStyles();

  return (
    <Container className={classes.root}>
      <Box>
        {composition?.workExample?.map(item => (
          // TODO: use isActiveCampaign & progress from CE data (once available)
          <SelectScoreItem 
            item={item} 
            isActiveCampaign={false}
            progress={60}
            onItemClick={() => onLoadScore(item)}
          />
        ))}
      </Box>
    </Container>
  );
}
SelectScoreModal.propTypes = {
  composition: PropTypes.object,
  onLoadScore: PropTypes.func,
};