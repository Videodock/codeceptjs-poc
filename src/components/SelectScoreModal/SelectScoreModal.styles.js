import { createStyles } from '@material-ui/styles';

export default ({ spacing, palette, shape, breakpoints, typography }) => createStyles({
  root: {
  },
});
