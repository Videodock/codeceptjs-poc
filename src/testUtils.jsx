import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
