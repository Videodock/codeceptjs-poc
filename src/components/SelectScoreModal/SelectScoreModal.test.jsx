import React from 'react';
import { render } from '../../testUtils';
import SelectScoreModal from './SelectScoreModal';

describe('<SelectScoreModal />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<SelectScoreModal />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
