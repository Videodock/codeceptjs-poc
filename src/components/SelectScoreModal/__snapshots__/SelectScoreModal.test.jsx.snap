// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectScoreModal /> renders and matches snapshot 1`] = `
<div
  class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
>
  <div
    class="MuiBox-root MuiBox-root-11"
  />
</div>
`;
